pydantic/main.py

```diff
@@ -767,6 +769,17 @@ def __getattr__(self, item: str) -> Any:
                         # this is the current error
                         raise AttributeError(f'{type(self).__name__!r} object has no attribute {item!r}')
 
+        def __getattribute__(self, item: str) -> Any:
+            try:
+                deprecated_inst = type(self).__pydantic_deprecated_fields__[item]
+                if deprecated_inst.category is not None:
+                    warnings.warn(
+                        deprecated_inst.message, deprecated_inst.category, stacklevel=deprecated_inst.stacklevel
+                    )
+            except (KeyError, TypeError):  # `TypeError` raised if `BaseModel` instantiated directly.
+                pass
+            return object.__getattribute__(self, item)
```

Comment:
With the `__getattribute__` method defined, the following script runs about 7x slower than without:
```python
class MyModel(BaseModel):
    x: int

my_model = MyModel(x=1)

t0 = time.time()
for _ in range(100_000):
    my_model.x
t1 = time.time()
print(t1 - t0)
```
even if I change it to actually do something, such as `my_model.x += 1` it is still ~4x slower.

I think that's probably too much of a performance price to pay for non-users of this feature for us to merge this as is.

I think we need to either find a way to implement this without the use of `__getattribute__` (ideally through the use of a descriptor for the deprecated fields), or at _least_ to make sure that the `__getattribute__` is only defined when there are deprecated fields on the model.

Reply:
Ouch, I was a bit worried with the usage of `__getattribute__` but 7x is a lot. I'll try to find an alternative solution!